helpers.lua

```diff
@@ -267,96 +267,89 @@ end
-        return true
+    -- If no permission matches, it means that there is no permission defined for this url.
+    if longest_permission_match == "" then
+        logger.debug("No access rules defined for user "..user..", assuming it cannot access.")
+        return false
     end
 
-    -- Loop through user's ACLs and return if the URL is authorized.
-    allowed_url_matches = {}
-    for url, app in pairs(conf["users"][user]) do
+    -- All user in this permission
+    allowed_users = conf["permissions"][longest_permission_match]
 
-        -- Replace the original domain by a local one if you are connected from
-        -- a non-global domain name.
-        if ngx.var.host == conf["local_portal_domain"] then
-            url = string.gsub(url, conf["original_portal_domain"], conf["local_portal_domain"])
+    -- The user has permission to access the content if he is in the list of this one
+    if allowed_users then
+        for _, u in pairs(allowed_users) do
+            if u == user then
+                logger.debug("User "..user.." can access "..ngx.var.uri)
+                log_access(user, longest_permission_match)
+                return true
+            end
         end
+    end
 
-        if string.ends(url, "/") then
-            url = string.sub(url, 1, -1)
-        end
+    logger.debug("User "..user.." cannot access "..ngx.var.uri)
+    return false
+end
 
-        if string.starts(ngx.var.host..ngx.var.uri, url) then
-            logger.debug("User is allowed to access this match : "..url)
-            table.insert(allowed_url_matches,url)
-        end
+function permission_matches()
+    if not conf["permissions"] then
+        conf["permissions"] = {}
     end
 
-    -- Keep only the longest match and compare it to the longest protected
-    -- match e.g. we don't want to allow the user to access /foo/admin if
-    -- /foo/admin is protected, but this user is only allowed to access /foo
-    local longest_allowed_match = longest_url_path(allowed_url_matches) or ""
-    local longest_protected_match = longest_url_path(protected_matches()) or ""
-
-    logger.debug("Longest allowed match : "..longest_allowed_match)
-    logger.debug("Longest protected match : "..longest_protected_match)
-
-    -- For the user to be able to access the content, at least one rule should
-    -- exist and it should be the longest match
-    if longest_allowed_match ~= ""
-    and string.len(longest_allowed_match) >= string.len(longest_protected_match) then
-        logger.debug("Logged-in user can access "..ngx.var.uri)
-        log_access(user, longest_allowed_match)
-        return true
-    else
-        logger.debug("Logged-in user cannot access "..ngx.var.uri)
-        return false
+    local url_matches = {}
+
+    for url, permission in pairs(conf["permissions"]) do
+        if string.starts(ngx.var.host..ngx.var.uri..uri_args_string(), url)
+        or string.starts(ngx.var.uri..uri_args_string(), url) then
```

Comment:
As I understand here you check here only on the path without the domain. I'm a little bit confused about that. Does it mean that if a packager create a permission with an url `/abcd`. The url will be accessible from all yunohost domain ?

Reply:
This line is a bad copy/pasta from other rules.